### ISSUE TITLE
✨ [REQUEST] Mealie Ingress Support

### ISSUE THREAD

**Issue by minmaxat** [alexbelgium/hassio-addons#1061]
### Checked

- [X] If a new addon, I have checked on  Google that such as addon doesn't already exists
- [X] If a new feature, I have searched in the discussions & issues that it was not already covered

### Which addon?

Mealie

### Is your feature request related to a problem? Please describe

it would be great to Access Mealie through NabuCasa URL / Ingress

### Describe the solution you'd like

it would be great to Access Mealie through NabuCasa URL / Ingress

**Comment by nickrout** [comment]
Just a +1

**Comment by alexbelgium** [comment]
Hi, sorry this is not supported by the app : https://nightly.mealie.io/documentation/getting-started/faq/#can-i-serve-mealie-on-a-subpath

**Comment by joostlek** [comment]
Is there a way to rewrite the paths?

**Comment by alexbelgium** [comment]
Hi actually easiest is to use an iframe. It will work straight away! As long as you have exposed the mealie port through your router : https://www.home-assistant.io/integrations/panel_iframe/

**Comment by alexbelgium** [comment]
I've given up after lots of tries some time again experimenting with nginx commands that modify the input such as redirect or sub_filter, each time modifying the nginx config of my mealie test addon using portainer, and never could find something that allowed to work. I haven't tried using auth through nginx though, perhaps it could help!

**Comment by joostlek** [comment]
But what's the reason why it doesn't work? Does it try to resolve the URLs absolute?

**Comment by alexbelgium** [comment]
that's my reminder indeed - that everything is hardcoded to /, I had a discussion with the dev that led to the same conclusion that sub path are not supported

**Comment by mountwebs** [comment]
> Hi actually easiest is to use an iframe. It will work straight away! As long as you have exposed the mealie port through your router : https://www.home-assistant.io/integrations/panel_iframe/

@alexbelgium Would this also work with nabu casa?

Too bad it isn't possible to use Mealie over ingress!

**Comment by alexbelgium** [comment]
> > Hi actually easiest is to use an iframe. It will work straight away! As long as you have exposed the mealie port through your router : https://www.home-assistant.io/integrations/panel_iframe/
> 
> @alexbelgium Would this also work with nabu casa?
> 
> Too bad it isn't possible to use Mealie over ingress!

Hi, it should as long as nabu casa allows to map external ports as a domain name would ! It works with my free self hosted domain at least

**Comment by joostlek** [comment]
AFAIK it doesn't. HA cloud doesn't even use the normal way of calling the http server (that's what I experienced from working with cloudhooks), so I don't expect they can connect to a separate port

**Comment by alexbelgium** [comment]
You could try with a free (or paid) domain also instead of nabu casa

**Comment by alexbelgium** [comment]
Me neither but there are solutions such as duckdns i think to dynamically map your IP to domain

**Comment by joostlek** [comment]
Let me rephrase, I don't own my public IP. I live in an apartment building where there is central routing, thus I can't open ports

**Comment by itn3rd77** [comment]
@alexbelgium I don't want to ride a dead horse. But ingress would be realy nice and I just found this post:
https://github.com/mealie-recipes/mealie/discussions/695#discussioncomment-9888004

Does that help?

**Comment by alexbelgium** [comment]
Hi, the proposed solution will not work `

ENV SUB_PATH=$SUB_PATH # this environment var will only exist during
docker build time

`

Alas the ingress url is defined not during the build time, but only after
the add-on is installed. You would have a much easier time to install
Mealie with iframe to have it in the sidebar

**Comment by alexbelgium** [comment]
Ingress is now implemented! Please let me know how that goes
